sdk/src/main/java/com/mapbox/maps/renderer/widget/BitmapWidget.kt

```diff
@@ -8,24 +8,84 @@ import com.mapbox.maps.MapboxExperimental
  *
  * @param bitmap bitmap used to draw widget
  * @param position position of widget
- * @param marginX horizontal margin in pixels
- * @param marginY vertical margin in pixels
  */
 @MapboxExperimental
 open class BitmapWidget @JvmOverloads constructor(
   bitmap: Bitmap,
-  position: WidgetPosition = WidgetPosition(
-    vertical = WidgetPosition.Vertical.TOP,
-    horizontal = WidgetPosition.Horizontal.LEFT,
-  ),
-  marginX: Float = 0f,
-  marginY: Float = 0f,
+  position: WidgetPosition = WidgetPosition {
+    verticalAlignment = WidgetPosition.Vertical.TOP
+    horizontalAlignment = WidgetPosition.Horizontal.LEFT
+    offsetX = 0f
+    offsetY = 0f
+  }
 ) : Widget() {
+  /**
+   * The deprecated constructor for BitmapWidget.
+   *
+   * @param bitmap bitmap used to draw widget
+   * @param position position of widget
+   * @param marginX horizontal margin in pixels
+   * @param marginY vertical margin in pixels
+   */
+  @Deprecated(
+    message = "Constructor with margins is deprecated, the offset parameters has been merged into " +
+      "the WidgetPosition class, and the legacy constructor might be removed in future releases.",
+    replaceWith = ReplaceWith("BitmapWidget(context, position)")
+  )
+  constructor(
+    bitmap: Bitmap,
+    position: WidgetPosition = WidgetPosition(
+      vertical = WidgetPosition.Vertical.TOP,
+      horizontal = WidgetPosition.Horizontal.LEFT,
+    ),
+    marginX: Float = 0f,
+    marginY: Float = 0f,
+  ) : this(
+    bitmap = bitmap,
+    position = WidgetPosition {
+      horizontalAlignment = position.horizontalAlignment
+      verticalAlignment = position.verticalAlignment
+      offsetX = when (position.horizontalAlignment) {
+        // as the WidgetPosition.offsetX now uses the absolute direction towards the right of the
+        // screen, and for marginX we move the widget towards opposite of the horizontal alignment,
+        // we need to flip the sign when horizontal alignment is set to RIGHT.
+        WidgetPosition.Horizontal.LEFT, WidgetPosition.Horizontal.CENTER -> marginX
```

Comment:
should `WidgetPosition.Horizontal.CENTER` return `0` value? or `left` one

Reply:
I ported the logic from the initial `setTranslation` implementation, we shouldn't change the behaviour here.